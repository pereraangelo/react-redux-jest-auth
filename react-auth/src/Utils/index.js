import moment from "moment";

export const getGreetingTime = () => {
  let currentTime = new Date();

  const splitAfternoon = 12; // 24hr time to split the afternoon
  const splitEvening = 15; // 24hr time to split the evening
  const currentHour = moment(currentTime).format("HH");

  if (currentHour >= splitAfternoon && currentHour <= splitEvening) {
    // Between 12 PM and 5PM
    return "Good Afternoon";
  } else if (currentHour >= splitEvening) {
    // Between 5PM and Midnight
    return "Good Evening";
  }
  // Between Dawn and noon
  return "Good Morning";
};



export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const simulateChangeOnInput = (wrapper, inputSelector, newValue) => {
  let input = findByTestAtrr(wrapper, inputSelector);
  input.simulate("change", {
    target: { name: inputSelector, value: newValue },
  });

  return findByTestAtrr(wrapper, inputSelector);
};


export const simulateCheckBox = (wrapper, inputSelector, newValue) => {
    let input = findByTestAtrr(wrapper, inputSelector);
    input.simulate('change', {target: {name:inputSelector,checked: newValue}});
  
    return findByTestAtrr(wrapper, inputSelector);
  };

  