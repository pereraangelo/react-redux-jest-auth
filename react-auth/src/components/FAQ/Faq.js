import React from "react";
import { Link } from "react-router-dom";

const data = {
  title: "FAQ ",
  rows: [
    {
      id:1,
      title: "Lorem ipsum dolor sit amet,",
      content: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. In sed tempor sem. Aenean vel turpis feugiat,
              ultricies metus at, consequat velit. Curabitur est nibh, varius in tellus nec, mattis pulvinar metus.
              In maximus cursus lorem, nec laoreet velit eleifend vel. Ut aliquet mauris tortor, sed egestas libero interdum vitae.
              Fusce sed commodo purus, at tempus turpis.`,
    },
    {
      id:2,
      title: "Nunc maximus, magna at ultricies elementum",
      content:
        "Nunc maximus, magna at ultricies elementum, risus turpis vulputate quam, vitae convallis ex tortor sed dolor.",
    },
    {
      id:3,
      title: "Curabitur laoreet, mauris vel blandit fringilla",
      content: `Curabitur laoreet, mauris vel blandit fringilla, leo elit rhoncus nunc, ac sagittis leo elit vel lorem.
            Fusce tempor lacus ut libero posuere viverra. Nunc velit dolor, tincidunt at varius vel, laoreet vel quam.
            Sed dolor urna, lobortis in arcu auctor, tincidunt mattis ante. Vivamus venenatis ultricies nibh in volutpat.
            Cras eu metus quis leo vestibulum feugiat nec sagittis lacus.Mauris vulputate arcu sed massa euismod dignissim. `,
    },
    {
      id:4,
      title: "What is the package version",
      content: <p>current version is 1.2.1</p>,
    },
  ],
};

function Faq(props) {
  return (  
    <div className="FAQPage">
      <div className="mb-5">
        Go back to
        <Link to="/" className=" btn-link m-3" data-test="login">
          {" "}
          Login{" "}
        </Link>
        <Link to="/registration" className=" btn-link pr-0" data-test="registration">
          {" "}
          Registration{" "}
        </Link>
      </div>

      <h1>FAQ </h1>
      <div>
        {data.rows.map((faq) => (
          <div key={faq.id} className=" p-5">
            <h3 className="font-weight-bold"> {faq.title}</h3>
            <span className="text-justify ">{faq.content}</span>
          </div>
        ))}
      </div>
    </div>
  );
}

export default Faq;
