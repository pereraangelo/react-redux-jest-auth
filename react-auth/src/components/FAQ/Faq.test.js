import React from "react";
import ReactDOM from 'react-dom'
import {shallow} from "enzyme"; 
import Faq from "./Faq";
import {MemoryRouter} from "react-router";
import { findByTestAtrr } from "../../Utils";
import renderer from 'react-test-renderer'




describe('FAQ component',()=>{ 
    const wrapper = shallow( <Faq />);
    it('Faq commponent matches the snapshot',()=>{
        const tree=renderer.create(<MemoryRouter ><Faq/></MemoryRouter>).toJSON()
        expect(tree).toMatchSnapshot()
    });
      
 
    it('Faq page should load without crahing',()=>{
        const div = document.createElement('div');
        ReactDOM.render(<MemoryRouter ><Faq/></MemoryRouter>, div);
    });
    
    it('redirect user to registration page',()=>{
        const faq = findByTestAtrr(wrapper, 'registration').props().to
        expect(faq).toEqual("/registration")
         
    })


    it('redirect user to Login page',()=>{
        const cancle = findByTestAtrr(wrapper, 'login').props().to
        
        expect(cancle).toEqual("/") 
         
    })

})

