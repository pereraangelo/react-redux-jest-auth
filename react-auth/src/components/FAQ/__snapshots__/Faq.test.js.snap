// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ component Faq commponent matches the snapshot 1`] = `
<div
  className="FAQPage"
>
  <div
    className="mb-5"
  >
    Go back to
    <a
      className=" btn-link m-3"
      data-test="login"
      href="/"
      onClick={[Function]}
    >
       
      Login
       
    </a>
    <a
      className=" btn-link pr-0"
      data-test="registration"
      href="/registration"
      onClick={[Function]}
    >
       
      Registration
       
    </a>
  </div>
  <h1>
    FAQ 
  </h1>
  <div>
    <div
      className=" p-5"
    >
      <h3
        className="font-weight-bold"
      >
         
        Lorem ipsum dolor sit amet,
      </h3>
      <span
        className="text-justify "
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. In sed tempor sem. Aenean vel turpis feugiat,
              ultricies metus at, consequat velit. Curabitur est nibh, varius in tellus nec, mattis pulvinar metus.
              In maximus cursus lorem, nec laoreet velit eleifend vel. Ut aliquet mauris tortor, sed egestas libero interdum vitae.
              Fusce sed commodo purus, at tempus turpis.
      </span>
    </div>
    <div
      className=" p-5"
    >
      <h3
        className="font-weight-bold"
      >
         
        Nunc maximus, magna at ultricies elementum
      </h3>
      <span
        className="text-justify "
      >
        Nunc maximus, magna at ultricies elementum, risus turpis vulputate quam, vitae convallis ex tortor sed dolor.
      </span>
    </div>
    <div
      className=" p-5"
    >
      <h3
        className="font-weight-bold"
      >
         
        Curabitur laoreet, mauris vel blandit fringilla
      </h3>
      <span
        className="text-justify "
      >
        Curabitur laoreet, mauris vel blandit fringilla, leo elit rhoncus nunc, ac sagittis leo elit vel lorem.
            Fusce tempor lacus ut libero posuere viverra. Nunc velit dolor, tincidunt at varius vel, laoreet vel quam.
            Sed dolor urna, lobortis in arcu auctor, tincidunt mattis ante. Vivamus venenatis ultricies nibh in volutpat.
            Cras eu metus quis leo vestibulum feugiat nec sagittis lacus.Mauris vulputate arcu sed massa euismod dignissim. 
      </span>
    </div>
    <div
      className=" p-5"
    >
      <h3
        className="font-weight-bold"
      >
         
        What is the package version
      </h3>
      <span
        className="text-justify "
      >
        <p>
          current version is 1.2.1
        </p>
      </span>
    </div>
  </div>
</div>
`;
